astropy/io/misc/yaml.py

```diff
@@ -222,6 +223,19 @@ def ignore_aliases(self, data):
 AstropyDumper.add_representer(Time, _time_representer)
 AstropyDumper.add_representer(TimeDelta, _timedelta_representer)
 AstropyDumper.add_representer(coords.SkyCoord, _skycoord_representer)
+AstropyDumper.add_representer(np.int32,
+                              yaml.representer.Representer.represent_int)
+try:
+    AstropyDumper.add_representer(np.int64,
+                                  yaml.representer.Representer.represent_long)
+except:
+    warnings.warn("This version of Pyyaml does not support long ints; using "
```

Comment:
I would not have this warning emitted here because it means you will always get the warning on importing the module regardless of actually needing long int.

Can you clarify (maybe with an example) of why you would need to represent a numpy 64-bit int with `represent_long` instead of int?  From the spec I don't see why you would want to specifically use `long` in this case since `int` works:  http://yaml.org/spec/1.2/spec.html#id2803828.